methods.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_map(proc = nil)
+    temp = []
+    i = 0
+    if proc.nil?
+      length.times do
+        temp.push(yield(self[i]))
+        i += 1
+      end
+    else
+      length.times do
+        temp.push(proc.call(self[i]))
+        i += 1
+      end
+    end
+    temp
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    length.times do
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    length.times do
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    length.times do
+      return true if yield(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    i = 0
+    length.times do
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    unless arg.nil?
+      i = 0
+      count = 0
+      length.times do
+        count += 1 if self[i] == arg
+        i += 1
+      end
+      return count
+    end
+    length
+  end
```

Comment:
- [ ] `my_count` is not working as the original `count` method, also is not passing all test cases. 

### `count` is an Enumerable that has these characteristics:
- Returns the count of elements inside an array/range/hash.
- When a block is given it returns the count of elements that match the condition
```
(1..3).count { |num| num > 2 }
# 1
```
[Enumerable#count](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-count)